core/src/main/kotlin/org/lflang/generator/ts/TSActionGenerator.kt

```diff
@@ -45,12 +45,21 @@ class TSActionGenerator(
                         ", " + action.minDelay.toTsTime()
                     }
                 }
-                if (action.name in networkMessageActions) {
+                // if (action.name in networkMessageActions) {
+                //     actionInstantiations.add(
+                //         "this.${action.name} = new __FederatePortAction<${action.tsActionType}>($actionArgs);")
+                // } else {
+                //     actionInstantiations.add(
+                //         "this.${action.name} = new __Action<${action.tsActionType}>($actionArgs);")
+                // }
+                if (action.name.take(7) == "network") {
```

Comment:
Let's not forget to clean this up so that if the user creates an action that starts with "network", it isn't interpreted as a a generated action that is part of the federated infrastructure. A magic attribute such as "__federatePortAction` might be sufficient to mark these.

It is a long-standing goal to do less ad-hoc string matching deep in the code generators. There are several other instances of string matching here in places where I do not think we should be doing string matching

Reply:
> if the user creates an action that starts with "network", it isn't interpreted as a generated action that is part of the federated infrastructure.

Oh, I didn't notice this. You're right. However, I don't know how to classify normal `Action` and `FederatePortAction`. Because the `Action` interface only has its name, origin, and policy. Do we have to add a new attribute in lf files?

Reply:
Well, I'm not sure I remember all the details, and I'm not 100% sure that my suggestion will work, but an example of adding an attribute is [in this commit](https://github.com/lf-lang/lingua-franca/pull/1684/commits/b4a76297b444c4a317bc0aec03a2025ee6fe4538#diff-0a51816c8e589e8f0b45b48eb7857be7c6c45006c00dbbccb813d78939399ae0), where I added the `_tpoLevel` attribute. If you do not get around to fixing this today, then I can work on it overnight since I do not think it will be too important or complicated.

Reply:
I just added a new attribute named `_NetworkReactor` on network reactors instead of `_FederatePortAction` on federate port actions. Because the code generator needs to classify network sender and receiver reactors as well as normal `Action` and `FederatePortAction`. We can know that an action is `FederatePortAction` when it is contained by a network receiver.

Reply:
Great, thanks!